python_ta/contracts/__init__.py

```diff
@@ -17,13 +17,32 @@ def check_contracts(wrapped, instance, args, kwargs):
         return check_function_contracts(wrapped, instance, args, kwargs)
 
 
+def instance_method_wrapper(wrapped, rep_invariants=None):
+    if rep_invariants is None:
+        return check_contracts
+
+    @wrapt.decorator
+    def wrapper(wrapped, instance, args, kwargs):
+        r = check_function_contracts(wrapped, instance, args, kwargs)
+        init = getattr(instance, "__init__")
+        check_invariants(instance, rep_invariants, init.__globals__)
+        return r
+
+    return wrapper(wrapped)
+
+
 def add_class_contracts(wrapped, args, kwargs):
-    rep_invariants = parse_assertions(
-        wrapped.__doc__ or '', parse_token="Representation Invariant")
+    rep_invariants = []
+
+    # Iterating over all inhertied classes except Object
+    for cls in wrapped.__mro__[:-1]:
+        rep_invariants.extend(parse_assertions(
```

Comment:
This is a natural place to check whether `cls` already has a `__representation_invariants__` attribute.

Reply:
@ihasan98 please add a `__dict__` check here as well

Reply:
Changed it to sets